sources/packages/web/src/views/aest/institution/ApproveDenyDesignation.vue

```diff
@@ -36,56 +78,82 @@ import {
   UpdateDesignationDto,
   DesignationAgreementStatus,
 } from "@/types/contracts/DesignationAgreementContract";
-import { useModalDialog } from "@/composables";
-import { computed } from "vue";
+import { useModalDialog, useRules } from "@/composables";
 import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
-import { Role } from "@/types";
+import { Role, VForm } from "@/types";
+import { ref, reactive } from "vue";
+import ErrorSummary from "@/components/generic/ErrorSummary.vue";
 
 export default {
-  components: { ModalDialogBase, CheckPermissionRole },
-  props: {
-    designation: {
-      type: Object,
-      required: true,
-    },
-  },
-  setup(props: any) {
-    const { showDialog, showModal, resolvePromise } = useModalDialog<
-      UpdateDesignationDto | boolean
-    >();
-    let formData: any = undefined;
+  components: { ModalDialogBase, CheckPermissionRole, ErrorSummary },
+  setup() {
+    const {
+      showDialog,
+      resolvePromise,
+      showModal: showModalInternal,
+    } = useModalDialog<UpdateDesignationDto | boolean>();
+    const approveDenyDesignation = ref({} as VForm);
+    const formModel = reactive({} as UpdateDesignationDto);
+    const title = ref("");
+    const subTitle = ref("");
+    const submitLabel = ref("");
+    const showApprovalContent = ref(false);
+    const { checkNotesLengthRule, checkStringDateFormatRule } = useRules();
 
-    const title = computed(() =>
-      props.designation.designationStatus ===
-      DesignationAgreementStatus.Approved
-        ? "Approve Designation"
-        : "Decline Designation",
-    );
+    //Setting the title values based on the DesignationAgreementStatus status passed to show modal.
+    const showModal = async (designation: UpdateDesignationDto) => {
+      formModel.note = "";
+      formModel.designationStatus = designation.designationStatus;
+      formModel.locationsDesignations = designation.locationsDesignations;
 
-    const dialogClosed = () => {
-      resolvePromise(false);
-    };
+      if (
+        designation.designationStatus === DesignationAgreementStatus.Approved
+      ) {
+        title.value = "Approve designation";
```

Comment:
When I built this component it was built inside view, but it has to go inside components

Reply:
Thanks for moving the the file to components. 👍